src/gluonts/dataset/common.py

```diff
@@ -185,26 +186,39 @@ class FileDataset(Dataset):
         Must be a valid Pandas frequency.
     one_dim_target
         Whether to accept only univariate target time series.
+    cache
+        Indicates whether the dataset should be cached or not.
     """
 
     def __init__(
-        self, path: Path, freq: str, one_dim_target: bool = True,
+        self,
+        path: Path,
+        freq: str,
+        one_dim_target: bool = True,
+        cache: Optional[bool] = False,
```

Comment:
How is this optional? Shouldn’t this be either True or False?

I think Optional and a default None should be used in two cases:
1. Having None as value for an argument has some meaning, like “don’t use this feature”
2. The default value for the argument requires a complex expression and potentially depends on other arguments values, so one defaults it to None and sets it appropriately in the constructor body

Reply:
You are right, I think I have over used it.